JOSESwift/Sources/CryptoImplementation/RSA.swift

```diff
@@ -50,24 +50,42 @@ fileprivate extension AsymmetricKeyAlgorithm {
         switch self {
         case .RSA1_5:
             return .rsaEncryptionPKCS1
+        case .RSAES_OAEP:
+            return .rsaEncryptionOAEPSHA1
         }
     }
 
     /// Checks if the plain text length does not exceed the maximum
     /// for the chosen algorithm and the corresponding public key.
     func isPlainTextLengthSatisfied(_ plainText: Data, for publicKey: SecKey) -> Bool {
+        let mLen: Int = plainText.count
+        let k = SecKeyGetBlockSize(publicKey)
         switch self {
         case .RSA1_5:
             // For detailed information about the allowed plain text length for RSAES-PKCS1-v1_5,
-            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2).
-            return plainText.count <= (SecKeyGetBlockSize(publicKey) - 11)
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2.1).
+            return mLen <= (k - 11)
+        case .RSAES_OAEP:
+            // For detailed information about the allowed plain text length for RSAES_OAEP,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.1.1).
+            // hLen = input limitation for the hash function (2^61 - 1 octets for SHA-1)
+            let hLen = (1 << 61) - 1
```

Comment:
I think `hLen` for SHA-1 is 20 bytes (160 bits). `(1 << 61) - 1` is the [input limitation](https://tools.ietf.org/html/rfc3447#section-7.1.1) of SHA-1. Please correct me if I got it wrong.

Reply:
Fixed